### PR TITLE
dev-util/makeheaders: bump, add myself as a maintainer

### PR DESCRIPTION
**Pull request by BratishkaErik**
Extracted from https://github.com/gentoo/gentoo/pull/28423
Confirmed to work with Clang 16 without changes

### CODE REVIEW

**Comment by BratishkaErik** [comment]
ping

File: dev-util/makeheaders/makeheaders-20220411.ebuild [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~sparc ~x86"
+
+S="${WORKDIR}"
+
+src_compile() {
+	$(tc-getCC) ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} -o makeheaders "${DISTDIR}"/makeheaders-${PV}.c || die
```
Comment (ceamac) [inline dev-util/makeheaders/makeheaders-20220411.ebuild:27]:
```suggestion
	$(tc-getCC) ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} -o makeheaders makeheaders-${PV}.c || die
```
We don't compile stuff directly from `${DISTDIR}`.  You can write a `src_unpack` function to copy all files from `${A}` to `${S}`, then you will have them available.

File: dev-util/makeheaders/makeheaders-20220411.ebuild [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+src_install() {
+	dobin makeheaders
+
+	local HTML_DOCS=( "${DISTDIR}"/makeheaders-${PV}.html )
```
Comment (ceamac) [inline dev-util/makeheaders/makeheaders-20220411.ebuild:33]: `HTML_DOCS` should be at global scope (and also without `${DISTDIR}`).

**Review by ceamac — CHANGES_REQUESTED** [review]

File: dev-util/makeheaders/makeheaders-20220411.ebuild [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~sparc ~x86"
+
+HTML_DOCS=( "makeheaders-${PV}.html" )
+
+src_unpack() {
+	mkdir "${S}" || die
+	cp "${DISTDIR}"/makeheaders-${PV}.c "${DISTDIR}"/makeheaders-${PV}.html "${S}" || die
```
Comment (ceamac) [inline dev-util/makeheaders/makeheaders-20220411.ebuild:28]: You could copy them without the version in the name, i.e., `"${S}"/makeheaders.{c,html}` - in this case the doc will have a normal name (the version is in the directory name anyway)

**Review by ceamac — CHANGES_REQUESTED** [review]
Also if you could consider changing the name of the ebuild (actually the version).  This is a snapshot release, if you name it *-2022mmdd* and later upstream will make an official release, say *0.1* or *1.0*, you will have the users downgrade the package.

Please read the [file naming rules](https://devmanual.gentoo.org/ebuild-writing/file-format/#file-naming-rules) and especially the part about [snapshot releases](https://devmanual.gentoo.org/ebuild-writing/file-format/#snapshots-and-live-ebuilds)

Since we already have a *-0_p4*, my suggestion would be to name it *-0_p20220411*

**Review by ceamac — APPROVED** [review]
Thank you for contributing

**Pull request closed without merging**